app/controllers/App.scala

```diff
@@ -35,14 +34,27 @@ class App(val wsClient: WSClient) extends Controller with PanDomainAuthActions {
     }
   }
 
-  def createAtom(atomType: String) = AuthAction {
+  def createAtom(atomType: String) = AuthAction { req =>
     APIResponse{
       for {
         atomType <- validateAtomType(atomType)
         ds <- AtomDataStores.getDataStore(atomType, Preview)
-        result <- AtomWorkshopDB.createAtom(ds, atomType)
+        result <- AtomWorkshopDB.createAtom(ds, atomType, req.user)
       } yield AtomWorkshopAPIResponse("Atom creation successful")
     }
   }
 
+  def updateEntireAtom(atomType: String, id: String) = AuthAction { req =>
+    APIResponse {
+      for {
+        atomType <- validateAtomType(atomType)
+        payload <- extractRequestBody(req.body.asText)
+        newAtom <- parseAtomJson(payload)
+        datastore <- AtomDataStores.getDataStore(atomType, Preview)
+        currentAtom <- AtomWorkshopDB.getAtom(datastore, atomType, id)
+        result <- AtomWorkshopDB.updateAtom(datastore, atomType, req.user, currentAtom,newAtom)
```

Comment:
I wonder if we should be returning the atom in the case of create and update? I realise this would require a change to the atom-maker library so I don't think it should be blocker to getting this out I'm just curious why the decision was made to return unit...